zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/AdHocSubProcessTransformer.java

```diff
@@ -93,10 +93,16 @@ private static void setAdHocActivities(
 
   private static void setImplementationType(
       final ExecutableAdHocSubProcess executableAdHocSubProcess, final AdHocSubProcess element) {
+
     final var implementationType =
-        Optional.ofNullable(element.getSingleExtensionElement(ZeebeAdHoc.class))
-            .map(ZeebeAdHoc::getImplementationType)
+        Optional.ofNullable(element.getExtensionElements())
+            .map(
+                extensionElements ->
+                    extensionElements.getChildElementsByType(ZeebeTaskDefinition.class))
+            .filter(taskDefinitions -> !taskDefinitions.isEmpty())
+            .map(taskDefinitions -> ZeebeAdHocImplementationType.JOB_WORKER)
             .orElse(ZeebeAdHocImplementationType.BPMN);
```

Comment:
I did not add any specific tests for the transformation as it should be implicitly covered by `AdHocSubProcessTest` and `JobBasedAdHocSubProcessTest`, but happy to add specific transformer tests as well.